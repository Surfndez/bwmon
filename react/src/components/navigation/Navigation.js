import React from 'react';
import { Link } from 'react-router-dom';
import { makeStyles } from '@material-ui/core/styles';
import { BottomNavigation, BottomNavigationAction } from '@material-ui/core';
import useNavigation from './UseNavigation';

const useStyles = makeStyles({});

const Navigation = ({ menu }) => {
    const classes = useStyles();
    const { index, setIndex } = useNavigation(menu);

    const handleChange = (event, newRouteIndex) => setIndex(newRouteIndex);

    return (
        <BottomNavigation value={index} onChange={handleChange} showLabels className={classes.root}>
            {menu.map((route, index) => (
                <BottomNavigationAction
                    id={route.id}
                    key={route.pathname}
                    value={index}
                    label={route.label}
                    icon={route.icon}
                    component={Link}
                    to={route.pathname}
                />
            ))}
        </BottomNavigation>
    );
};

export default Navigation;
