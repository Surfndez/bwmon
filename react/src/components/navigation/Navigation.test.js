import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { create } from 'react-test-renderer';
import Navigation from './Navigation';

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useLocation: () => ({
        pathname: '/pathname2',
    }),
}));

describe('Navigation', () => {
    const menu = [
        { id: 'id1', pathname: '/pathname1', label: 'label1', icon: 'icon1' },
        { id: 'id2', pathname: '/pathname2', label: 'label2', icon: 'icon2' },
        { id: 'id3', pathname: '/pathname3', label: 'label3', icon: 'icon3' },
    ];

    test('renders correctly', () => {
        const tree = create(
            <BrowserRouter>
                <Navigation menu={menu} />
            </BrowserRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
