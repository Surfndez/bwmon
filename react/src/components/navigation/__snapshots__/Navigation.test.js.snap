// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renders correctly 1`] = `
<div
  className="MuiBottomNavigation-root"
>
  <a
    aria-disabled={false}
    className="MuiButtonBase-root MuiBottomNavigationAction-root Mui-selected"
    href="/pathname1"
    id="id1"
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <span
      className="MuiBottomNavigationAction-wrapper"
    >
      icon1
      <span
        className="MuiBottomNavigationAction-label Mui-selected"
      >
        label1
      </span>
    </span>
  </a>
  <a
    aria-disabled={false}
    className="MuiButtonBase-root MuiBottomNavigationAction-root"
    href="/pathname2"
    id="id2"
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <span
      className="MuiBottomNavigationAction-wrapper"
    >
      icon2
      <span
        className="MuiBottomNavigationAction-label"
      >
        label2
      </span>
    </span>
  </a>
  <a
    aria-disabled={false}
    className="MuiButtonBase-root MuiBottomNavigationAction-root"
    href="/pathname3"
    id="id3"
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <span
      className="MuiBottomNavigationAction-wrapper"
    >
      icon3
      <span
        className="MuiBottomNavigationAction-label"
      >
        label3
      </span>
    </span>
  </a>
</div>
`;
